mnefun/_mnefun.py

```diff
@@ -2330,3 +2330,117 @@ def plot_reconstruction(evoked, origin=(0., 0., 0.04)):
+    if buflen <= 0:
+        raise ValueError('Window length should be >0')
+    cfreqs = []
+    if (len(raw.info['hpi_meas']) > 0 and
+            'coil_freq' in raw.info['hpi_meas'][0]['hpi_coils'][0]):
+        for coil in raw.info['hpi_meas'][0]['hpi_coils']:
+            cfreqs.append(coil['coil_freq'][0])
+    else:
+        raise ValueError('Cannot determine cHPI frequencies'
+                         'from raw data info')
+    print('\nNominal cHPI frequencies: ', cfreqs, ' Hz')
+    print('Sampling frequency:', sfreq, 'Hz')
+    print('Using line freqs:', linefreqs, ' Hz')
+    print('Using buffers of', buflen, 'samples =', buflen/sfreq, 'seconds\n')
+
+    raw.info['bads'] = []  # make sure to load all channels
+    pick_meg = pick_types(raw.info, meg=True)
+    pick_mag = pick_types(raw.info, meg='mag')
+    pick_grad = pick_types(raw.info, meg='grad')
+    nchan = len(pick_meg)
+
+    # create general linear model for the data
+    t = np.linspace(0, buflen/sfreq, endpoint=False, num=buflen)
+    model = np.c_[t, np.ones(t.shape)]  # model slope and DC
+    # add sine and cosine term for each freq
+    for f in list(linefreqs) + cfreqs:
+        model = np.c_[model, np.cos(2*np.pi*f*t), np.sin(2*np.pi*f*t)]
+    inv_model = np.linalg.pinv(model)
+
+    # loop through MEG data, fit linear model and compute SNR at each window
+    if stop:
+        stop = int(stop*sfreq)
+    else:
+        stop = raw.n_times
+    # drop last buffer to avoid overrun
+    bufs = range(0, int(stop), buflen)[:-1]
+    tvec = np.array(bufs)/sfreq
+    snr_avg_grad = np.zeros([len(cfreqs), len(bufs)])
+    snr_avg_mag = np.zeros([len(cfreqs), len(bufs)])
+    resid_vars = np.zeros([nchan, len(bufs)])
+    for ind, buf0 in enumerate(bufs):
+        megbuf = raw[pick_meg, buf0:buf0+buflen][0].transpose()
+        coeffs = np.dot(inv_model, megbuf)
+        coeffs_hpi = coeffs[2+2*len(linefreqs):]
+        resid_vars[:, ind] = np.var(megbuf-np.dot(model, coeffs), 0)
+        # get total HPI amplitudes by combining sine and cosine terms
+        hpi_amps = np.sqrt(coeffs_hpi[0::2, :]**2 + coeffs_hpi[1::2, :]**2)
+        # divide average HPI power by average variance
+        snr_avg_grad[:, ind] = np.divide((hpi_amps**2/2)[:, pick_grad].mean(1),
+                                         resid_vars[pick_grad, ind].mean())
+        snr_avg_mag[:, ind] = np.divide((hpi_amps**2/2)[:, pick_mag].mean(1),
+                                        resid_vars[pick_mag, ind].mean())
+
+    plt.figure()
+    cfreqs_legend = [str(fre)+' Hz' for fre in cfreqs]
+
+    # order curve legends according to mean of data
+    sind = np.argsort(snr_avg_grad.mean(axis=1))[::-1]
+    lines1 = plt.plot(tvec, 10*np.log10(snr_avg_grad.transpose()))
+    plt.title('Mean cHPI power / mean residual variance (gradiometers)')
+    plt.legend(np.array(lines1)[sind], np.array(cfreqs_legend)[sind],
+               prop={'size': legend_fontsize})
+    plt.ylabel('SNR (dB)')
```

Comment:
...and `ax.set_ylabel(...)` or better `ax.set(ylabel=..., xlabel=..., xlim=..., ...)`